### PR TITLE
feat(CO-51): Frontend dungeon hero action

### PR DESCRIPTION
**Pull request by Galicz555**

### CODE REVIEW

File: src/app/models/hero.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Equipment } from 'src/app/models/equipment';
```
Comment (matyaspeterfi) [inline src/app/models/hero.ts]: Why do we need this?

Comment (Galicz555) [reply src/app/models/hero.ts]: solved

File: src/app/components/dungeon/dungeon.component.ts [hunk 2]
```diff
@@ -13,21 +15,37 @@ export class DungeonComponent {
...
-  showSelectedItem(item) {
+  public showSelectedItem(item) {
     this._selectedObstacle = item;
   }
 
+  public onProgress(event) {
+    this._events = event [0];
+    this._hero = event [1];
+    const kiscica = new Dungeon(event[2]);
+    this._dungeon = kiscica;
```
Comment (juhaszPeterDev) [inline src/app/components/dungeon/dungeon.component.ts]: even if you need this variable for unknown reasons, name it better :)

File: src/app/components/hero-action/hero-action.component.ts [hunk 3]
```diff
@@ -1,19 +1,112 @@
...
+      return 'Continue';
+    } else if (this.dungeon.removedObstacles === this.dungeon.obstacles.length) {
+      this._action = 'collect reward';
+      return 'CollectReward';
+    }
+  }
+
+  public changeState() {
+    if (this._toState === 'FinalWords' && !this._decider) {
+      this._decider = 'a';
```
Comment (juhaszPeterDev) [inline src/app/components/hero-action/hero-action.component.ts]: this variable usage doesnt seem right. cant _decider be a boolean? why give this unclean value to it?

File: src/app/services/dungeon-actions/dungeon-actions.service.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    return request;
+  }
+
+  public finalWords(heroId: number, finalWords: string) {
+    let headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    headers = headers.append('heroId', JSON.stringify(heroId));
+    headers = headers.append('words', finalWords);
```
Comment (juhaszPeterDev) [inline src/app/services/dungeon-actions/dungeon-actions.service.ts]: pls put this info in the request payload

File: src/app/components/dungeon/dungeon.component.ts (hunk 2 above)
Comment (matyaspeterfi) [reply src/app/components/dungeon/dungeon.component.ts]: my fault lol

Comment (Galicz555) [reply src/app/components/dungeon/dungeon.component.ts]: solved

File: src/app/components/hero-action/hero-action.component.ts (hunk 3 above)
Comment (Galicz555) [reply src/app/components/hero-action/hero-action.component.ts]: solved

File: src/app/services/dungeon-actions/dungeon-actions.service.ts (hunk 4 above)
Comment (Galicz555) [reply src/app/services/dungeon-actions/dungeon-actions.service.ts]: solved

**Pull request merged**